### PR TITLE
Muxer: increase Continuity Counter of PAT and PMT

### PR DESCRIPTION
**Pull request by aler9**
Hello, this library allowed [rtsp-simple-server](https://github.com/aler9/rtsp-simple-server) to convert seamlessly from RTSP to HLS during the last months, hence a big thank you from the entire community.

As you may already know, HLS is a video/audio streaming format that uses MPEG-TS segments, containing H264 and AAC.

The only change i had to do to generate valid HLS was hacking a couple of functions in order to increase the continuity counter of PAT and PMT after every delivery.

Although the specification is not clear enough about the continuity counter of PAT and PMT, when it is not increased, VLC starts printing these kind of errors:

```
[00007f80ec001180] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 0
[00007f80ec001180] ts demux error: libdvbpsi error (PSI decoder): TS duplicate (received 0, expected 1) for PID 4096
```

At the same time, hls.js (JS library that converts from MPEG-TS to browser-compatible fMP4) is unable to generate a valid video stream.

PAT and PMT buffers are reused in order to save RAM.

### CODE REVIEW

**Review by tmm1 — APPROVED** [review]

File: muxer.go [hunk 1]
```diff
@@ -285,16 +290,12 @@ func (m *Muxer) retransmitTables(force bool) (int, error) {
 func (m *Muxer) WriteTables() (int, error) {
 	bytesWritten := 0
 
-	if m.patBytes.Len() != m.packetSize {
-		if err := m.generatePAT(); err != nil {
-			return bytesWritten, err
-		}
+	if err := m.generatePAT(); err != nil {
```
Comment (asticode) [inline muxer.go:293]: What's the reasoning behind removing this condition ?

File: muxer.go [hunk 2]
```diff
@@ -285,16 +290,12 @@ func (m *Muxer) retransmitTables(force bool) (int, error) {
...
+		return bytesWritten, err
 	}
 
-	if m.pmtBytes.Len() != m.packetSize {
-		if err := m.generatePMT(); err != nil {
-			return bytesWritten, err
-		}
+	if err := m.generatePMT(); err != nil {
```
Comment (asticode) [inline muxer.go:297]: What's the reasoning behind removing this condition ?

**Review by asticode — CHANGES_REQUESTED** [review]
Nice PR, just need some clarifications

File: muxer.go (hunk 1 above)
Comment (aler9) [reply muxer.go:293]: Hello, to increase the continuity counter, PAT and PMT needs to be regenerated during every `WriteTables()`, therefore they can't be cached anymore. That condition allowed to cache PAT and PMT, this is not possible anymore.

**Comment by asticode** [comment]
FYI I've created a `v1.9.0` tag

Cheers

**Comment by barbashov** [comment]
It seems that this PR introduced a bug: now tables version increase every time tables are written. Which doesn't go along with specification AFAIR: we should increase tables version only if they change

**Comment by tmm1** [comment]
I noticed the same and had to disable it: https://github.com/tmm1/go-astits/commit/015601332bc30f086404681608e5e558e62fd1dd

**Comment by asticode** [comment]
What do you think about this fix : https://github.com/asticode/go-astits/pull/38

**Comment by aler9** [comment]
I didn't notice that the version number was increasing every time tables are written - my only aim was increasing the continuity counter, as the specification says. So thanks for the fix.

**Comment by barbashov** [comment]
No worries, bugs happen :)

**Comment by asticode** [comment]
@aler9 yeah no worries 👍

**Pull request merged**